terraform/context_validate_test.go

```diff
@@ -87,35 +87,28 @@ func TestContext2Validate_varMapOverrideOld(t *testing.T) {
 		},
 	}
 
-	c := testContext2(t, &ContextOpts{
+	_, diags := NewContext(&ContextOpts{
 		Config: m,
 		ProviderResolver: providers.ResolverFixed(
 			map[string]providers.Factory{
 				"aws": testProviderFuncFixed(p),
 			},
 		),
-		Variables: InputValues{
-			"foo.foo": &InputValue{
-				Value:      cty.StringVal("bar"),
-				SourceType: ValueFromCaller,
-			},
-		},
+		Variables: InputValues{},
 	})
-
-	diags := c.Validate()
 	if !diags.HasErrors() {
+		// Error should be: The input variable "provider_var" has not been assigned a value.
```

Comment:
Thank you for adding these, I'll be sure to start doing the same!

Reply:
Honestly, if I were writing these tests from scratch today I'd rather make the test code itself assert the error message, but I didn't want to mess _too_ much with existing tests here and distract from the main goal...